src/xtgeo/grid3d/_gridprop_roxapi.py

```diff
@@ -108,10 +113,15 @@ def _convert_to_xtgeo_prop(
 
 
 def export_prop_roxapi(
-    self, project, gname, pname, realisation=0, casting="unsafe"
-):  # pragma: no cover
+    self: GridProperty,
+    project: roxar.Project,
+    gname: str,
+    pname: str,
+    realisation: int = 0,
+    casting: Literal["no", "equiv", "safe", "same_kind", "unsafe"] | None = "unsafe",
```

Comment:
Not sure if this was a bit too deep 😄 .. I checked available options in the Numpy documentation. But now I see that mypy is complaining in the grid_property module where this casting has been set to "str". 
Should I just keep the type for casting a "str" here as well?

Reply:
I think it's fine to change the casting annotating in `grid_property` as well. The only downside perhaps is that we may have to update it at some point -- but it remains more precise than `str`.

(Make sure you rebase main into your branch if you update `grid_property` though, not sure if you have the latest changes 😄)

Reply:
Ok thanks, I will make the change 🙂